src/libutil/error.hh

```diff
@@ -86,6 +86,7 @@ void printCodeLines(std::ostream & out,
 struct Trace {
     std::shared_ptr<AbstractPos> pos;
     hintformat hint;
+    bool frame;
```

Comment:
What about "importance", or "scope" ? The thing is, I produce a lot of trace elements about details of the nix expressions. If this is done everywhere, it gets too long to be useful. but if I print details until I reach a broader scope trace (like a function call), I stop printing the details and only print function calls.

All this fuss happened because I considered that addErrorTrace was a local trace, but it makes more sense to have it also in the stack trace. It is extremely difficult to strike the right balance here, and this is just a naive attempt.

Also, frame kind of makes sense, because it corresponds to a trace about a stack frame (well, initially, before adding addErrorTrace). I know technically all of the expressions correspond to calling functions, but I have decided to consider builtins calls as normal expressions, including addErrorTrace. This is just a work of printing lots of expressions and getting them all look good.

To make progress on this, we need a collection of error-generating expressions. Then we can work on comparing them, and finding what is the best info to display. This PR only tries to add context to some coercions, not to fix all the traces at once.

Reply:
It seems this is going to be tech debt then.